### PR TITLE
Allow registering AWS integration with custom config from the UI

### PR DESCRIPTION
**Pull request by cw75**
## Describe your changes and why you are making these changes
See screenshots below:

Screen Shot 2023-03-07 at 1 28 35 AM
Screen Shot 2023-03-07 at 1 28 46 AM

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Comment by vsreekanti** [comment]
Hm, this is quite long now, and I'm not the biggest fan of modals that scroll. We should think about making this a multi-page set up. Something like this: 
1. First, you enter access credentials, you hit Next.
2. Then you get the ability to optionally specify any of the other fields.
3. Then you confirm.

**Comment by cw75** [comment]
@vsreekanti Yeah I like your suggestion. I need to look into how to implement the multi-page version though. Do you think it's a blocker to shipping? If not, I'll create a task and work on other higher priority tasks and then get back to this one.

**Comment by vsreekanti** [comment]
We're not shipping this PR in this release, right?

**Comment by vsreekanti** [comment]
I would not release the feature as-is, but I don't have a strong opinion on the order in which you make the PRs, if that's what you're asking.

**Comment by cw75** [comment]
@vsreekanti Yeah we're planning to ship the project 3/22 and I was asking if the multi-page design is a release blocker. You said "I would not release the feature as-is" so th answer seems to be "yes". I'll work this into my planning.

**Comment by vsreekanti** [comment]
Got it — yes, I wasn't sure if you were asking about merging this PR or releasing the feature. I'm fine with merging this PR, but I would prefer not to release the feature without the multi-page design.

**Comment by cw75** [comment]
@likawind let me know if the state management part makes sense. It seems to work fine during my testing.

File: src/ui/common/src/components/integrations/dialogs/awsDialog.tsx [hunk 1]
```diff
@@ -127,6 +237,15 @@ export const AWSDialog: React.FC<Props> = ({
       </Box>
       {value?.type === AWSCredentialType.AccessKey && accessKeyTab}
       {value?.type === AWSCredentialType.ConfigFilePath && configPathTab}
+      <Box sx={{ borderBottom: 1, borderColor: 'divider', mb: 2 }}>
+        <Tabs value={engineType} onChange={(_, value) => setEngineType(value)}>
```
Comment (likawind) [inline src/ui/common/src/components/integrations/dialogs/awsDialog.tsx:241]: how do you send this field to backend? It appears we didn't call 'onUpdateField' when this option is changed?

File: src/ui/common/src/components/integrations/dialogs/awsDialog.tsx [hunk 2]
```diff
@@ -111,6 +128,99 @@ export const AWSDialog: React.FC<Props> = ({
...
+      />
+      <IntegrationTextInputField
+        spellCheck={false}
+        required={false}
+        label="CPU node type"
+        description="The EC2 instance type of the CPU node group."
+        placeholder={K8sPlaceholders.cpu_node_type}
+        onChange={(event) => {
+          k8sConfigs['cpu_node_type'] = event.target.value;
+          onUpdateField('k8s_serialized', JSON.stringify(k8sConfigs));
```
Comment (likawind) [inline src/ui/common/src/components/integrations/dialogs/awsDialog.tsx:156]: did you verify if this is working smoothly? It's different from what we did in some other places (where we use an additional internal state to track), but I also don't see any issue with your way

Try to type fast, delete fast, and switch between various fields. See if there's any latency when you type, and see if the created integration's fields matches what's in the dialog box. If that works, we can go with this solution.

Comment (cw75) [reply src/ui/common/src/components/integrations/dialogs/awsDialog.tsx:156]: I think so yeah it's working.

File: src/ui/common/src/components/integrations/dialogs/awsDialog.tsx (hunk 1 above)
Comment (cw75) [reply src/ui/common/src/components/integrations/dialogs/awsDialog.tsx:241]: Do I need to send this to backend? This var is just to select which tab (we only have k8s for now) to show and it doesn't need to be sent to the server side.

Comment (likawind) [reply src/ui/common/src/components/integrations/dialogs/awsDialog.tsx:241]: I see, I thought it's a part of the config, makes sense now

**Comment by cw75** [comment]
@likawind could you stamp this if it looks good?

**Review by likawind — APPROVED** [review]
:shipit:

**Pull request merged**